framework/src/main/java/org/checkerframework/common/returnsreceiver/qual/UnknownThis.java

```diff
@@ -0,0 +1,21 @@
+package org.checkerframework.common.returnsreceiver.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.*;
+
+/**
+ * The top type for the Returns Receiver Checker's type system. Values of the annotated type might
+ * be the receiver ({@code this}) or might not. Programmers should rarely write this type.
+ *
+ * @checker_framework.manual #returns-receiver-checker Returns Receiver Checker
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
+@DefaultQualifierInHierarchy
+@SubtypeOf({})
+@QualifierForLiterals(LiteralKind.NULL)
```

Comment:
Is there a reason to use the top type for `null`, `Void`, or lower bounds? It would be more standard to use the bottom type for these locations.

Reply:
For `null`, we want to get an error for code like this (from a test):
```java
        @This B dontRetThis() {
            // :: error: return.type.incompatible
            return null;
        }
```
Without using the top type for `null` we don't get that error.

For `Void` and lower bounds, I am less sure, but I'm pretty confident weird errors showed up in large benchmarks that motivated these additions.  I can try to track those down and add corresponding unit tests.

Reply:
Looked into this.  Because we want to use the top type for `null`, we also need to use it for generic type lower bounds.  Otherwise, the `GenericNull` test case from the all-systems tests fails.  I could not find a case where using `@UnknownThis` for `Void` was required, so I removed that.